WooCommerce/Classes/ViewRelated/Orders/OrdersSplitViewWrapperController.swift

```diff
@@ -97,10 +97,33 @@ private extension OrdersSplitViewWrapperController {
             return
         }
 
-        let orderDetailsViewController = OrderDetailsViewController(viewModels: viewModels, currentIndex: currentIndex)
-        let orderDetailsNavigationController = WooNavigationController(rootViewController: orderDetailsViewController)
+        let orderDetailsViewController = OrderDetailsViewController(
+            viewModels: viewModels,
+            currentIndex: currentIndex,
+            switchDetailsHandler: { [weak self] viewModels, currentIndex, isSelectedManually, completion in
+                self?.handleSwitchingDetails(viewModels: viewModels,
+                                             currentIndex: currentIndex,
+                                             isSelectedManually: isSelectedManually,
+                                             onCompletion: completion)
+            })
+
+        // When showing an order with multiple order view models to support quick navigation (up and down arrows),
+        // subsequent order details should replace the top order details view controller to avoid multiple order details
+        // in the navigation stack.
```

Comment:
```suggestion
        // When navigating between orders using up and down arrows, each new Order Details screen shown
        // should replace the topViewController, to avoid having to tap back through several Order Details
        // screens in the navigation stack. The back button should always go to the Order List.
        // The up and down arrows are enabled when there is more than one item in `viewModels`.
```